### ISSUE TITLE
Extensible plugins

### ISSUE THREAD

**Issue by GarthDB** [remijs/remi#12]
This is probably more of a question than an issue, but I want to make an extensible plugin that other plugins can inherit (I'm duplicating a bunch of code between the plugins) is there a good way to go about this? I tried making the plugin a class but remi doesn't like registering that.

**Comment by zkochan** [comment]
You can use these two extensions for remi:
https://github.com/remijs/remi-decorate
https://github.com/remijs/remi-expose

They try to mimic the way hapi plugins expose stuff

**Comment by GarthDB** [comment]
So if plugins have a shared base of code (I'm mainly thinking something along the lines of before and after register functionality) I should add that to the `app`? I could use decorate to add via one of the plugins, but that doesn't really matter to me yet.

**Comment by GarthDB** [comment]
Thanks for the quick response btw @zkochan

**Comment by zkochan** [comment]
Yes, adding it to the app object is the way it can be achieved. Just be sure to place the plugins in the correct order.

**Comment by GarthDB** [comment]
I understand if I'm using plugins and decorate to add the functionality I'd have to use them in the right order.

Another question it looks like I have to use `app.root.someMethod()` to get the method, right?

**Comment by zkochan** [comment]
only if you want to access it in the same plugin in which you decorated the app with someMethod. In the next plugins it'll be available via `app.someMethod()`.
